### PR TITLE
libavcodec/qsvenc.c: Disable MFX_GOP_STRICT when encode adpative gop

### PR DESCRIPTION
**Pull request by wenbinc-Bin**
b_strategy, p_strategy, adaptive_i and adaptive_b cannot work with MFX_GOP_STRICT, so disable MFX_GOP_STRICT when these features are used.

Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>

### CODE REVIEW

File: libavcodec/qsvenc.c [hunk 1]
```diff
@@ -644,6 +644,12 @@ static int check_enc_param(AVCodecContext *avctx, QSVEncContext *q)
     return 1;
 }
 
+static int is_adaptive_gop(QSVEncContext *q) {
+    if (q->adaptive_b || q->adaptive_i || q->b_strategy || q->p_strategy)
```
Comment (xhaihao) [inline libavcodec/qsvenc.c]: The default value for adaptive_b / adaptive_i is -1

Comment (wenbinc-Bin) [reply libavcodec/qsvenc.c]: Oh, thanks. I will update it.

**Review by xhaihao — APPROVED** [review]
LGTM.

**Comment by wenbinc-Bin** [comment]
submit to upstream: https://patchwork.ffmpeg.org/project/ffmpeg/patch/20221116021338.502587-1-wenbin.chen@intel.com/

**Pull request closed without merging**